homeassistant/components/iotawatt/sensor.py

```diff
@@ -0,0 +1,96 @@
+"""Support for IoTaWatt Energy monitor."""
+from __future__ import annotations
+
+from iotawattpy.sensor import Sensor
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.core import callback
+from homeassistant.helpers import entity_registry, update_coordinator
+
+from .const import DOMAIN, ENTITY_DESCRIPTION_KEY_MAP
+from .coordinator import IotawattUpdater
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add sensors for passed config_entry in HA."""
+    coordinator: IotawattUpdater = hass.data[DOMAIN][config_entry.entry_id]
+    created = set()
+
+    @callback
+    def _create_entity(key: str) -> IotaWattSensor:
+        """Create a sensor entity."""
+        created.add(key)
+        return IotaWattSensor(
+            coordinator=coordinator,
+            key=key,
+            mac_address=coordinator.data["sensors"][key].hub_mac_address,
+            name=coordinator.data["sensors"][key].getName(),
+            entity_description=ENTITY_DESCRIPTION_KEY_MAP.get(
+                coordinator.data["sensors"][key].getUnit(),
+                SensorEntityDescription("base_sensor"),
+            ),
+        )
+
+    async_add_entities(_create_entity(key) for key in coordinator.data["sensors"])
+
+    @callback
+    def new_data_received():
+        """Check for new sensors."""
+        entities = [
+            _create_entity(key)
+            for key in coordinator.data["sensors"]
+            if key not in created
+        ]
+        if entities:
+            async_add_entities(entities)
+
+    coordinator.async_add_listener(new_data_received)
+
+
+class IotaWattSensor(update_coordinator.CoordinatorEntity, SensorEntity):
+    """Defines a IoTaWatt Energy Sensor."""
+
+    _attr_force_update = True
+    _attr_icon = "mdi:flash"
```

Comment:
This will override the icons set by the device class. I suggest we don't do that.
```suggestion
```

Reply:
I removed the override of the icon. There a couple of sensors that don't have a device class specified, such as ```Hz```, ```VAR```, and ```VARh```. Is is okay for these sensors to have the default icon of the eye?

Reply:
I will set the icon in the ```SensorEntityDescription``` for these sensors without a device class